lib/src/value/calculation.dart

```diff
@@ -158,6 +382,200 @@ class SassCalculation extends Value {
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value rem(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("rem", args);
+    }
+    _verifyCompatibleNumbers(args);
+
+    var result = dividend.modulo(modulus);
+    if ((modulus.value < 0 && dividend.value >= 0) ||
+        (modulus.value >= 0 && dividend.value < 0)) {
+      if (modulus.value.isInfinite) return dividend;
+      if (result is SassNumber && result.value == 0) return SassNumber(-0.0).coerceToMatch(result);
+      return result.minus(dividend);
+    }
+    return result;
+  }
+
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    strategyOrNumber = _simplify(strategyOrNumber);
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+
+    var args = (strategyOrNumber, numberOrStep, step);
+
+    switch (args) {
```

Comment:
```suggestion
    switch ((
      _simplify(strategyOrNumber),
      numberOrStep.andThen(_simplify),
      step.andThen(_simplify)
    )) {
```

Part of the value of a switch-over-tuple, especially in contexts where a given argument may mean different things depending on how many arguments are passed, is that it _doesn't_ require you to create a bunch of intermediate variables until you're in a case where you know for sure what meaning each variable has.

Reply:
Sounds good, refactored the `switch` block taking into account this, and the review comment to cover more edge case. Pls let me know your thoughts @nex3

Reply:
Had to declare `strategyOrNumberSimplified` to reuse it when returning the calculation, since I cannot refer to the switch argument because it would return a dynamic object. (Also tried casting it to an Object, but I'm still not able to use it.)